tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,757 @@
+            self.test_context,
+            self.redpanda,
+            self.topic,
+            msg_size=self.params.msg_size,
+            max_msgs=None,  # consume everything
+            max_throughput_mb=self.params.consume_rate_mb *
+            self.params.parallel_consumers,
+            nodes=[self.preallocated_nodes[0]],
+            debug_logs=True,
+            trace_logs=True)
+
+    def _create_group_consumer(self):
+        return KgoVerifierConsumerGroupConsumer(
+            self.test_context,
+            self.redpanda,
+            self.topic,
+            msg_size=self.params.msg_size,
+            max_msgs=self.params.msg_count,
+            readers=self.params.parallel_consumers,
+            max_throughput_mb=self.params.consume_rate_mb *
+            self.params.parallel_consumers,
+            nodes=[self.preallocated_nodes[0]],
+            debug_logs=True,
+            trace_logs=True)
+
+    @cluster(num_nodes=4)
+    @parametrize(restart_stress=True)
+    @parametrize(restart_stress=False)
+    def constant_throughput_prolonged_test(self, restart_stress):
+        """
+        Stress the tiered storage upload path on a multiple partitions
+        for a prolonged time. This ensures that there is no skips happen
+        or lost data occurs for additional amounts of time with retention
+        equals to -1. Which is effectively turns it off
+
+        :param restart_stress: if true, additionally restart nodes during
+                               writes, and waive throughput success conditions:
+                               in this mode we are checking for stability and
+                               data durability.
+        """
+        # Overide duration
+        self.params.target_runtime = 300
+        producer = self._create_producer()
+
+        self.logger.info(f"Producing {self.params.msg_count} msgs "
+                         f"({self.params.write_bytes} bytes)")
+        produce_start = time.time()
+        producer.start()
+
+        if restart_stress:
+            while producer.produce_status.acked < self.params.msg_count:
+                # delay stress start for 30 sec by default
+                time.sleep(self.params.target_stress_start)
+                # restart one with 5 min pause
+                # real world: simulates cluster update
+                for node in self.redpanda.nodes:
+                    time.sleep(self.params.target_stress_delay)
+                    self.redpanda.restart_nodes(
+                        [node],
+                        start_timeout=self.params.node_start_timeout,
+                        stop_timeout=self.params.node_stop_timeout)
+
+            # Wait for the cluster to recover after final restart,
+            # so that subsequent post-stress success conditions
+            # can count on their queries succeeding.
+            self.redpanda.wait_until(self.metadata_readable,
+                                     timeout_sec=60,
```

Comment:
I'm a bit confused by this condition. Metadata being readable doesn't really have much indication of whether the cluster is _actually_ healthy. Does `redpanda.healthy()` not quiesce quickly enough?

Reply:
This comes from the test I've used as an example. Could be removed I guess.